lib/src/user.dart

```diff
@@ -92,4 +94,68 @@ class User {
     await _channel
         .invokeMethod('setUserTag', {"attributeKey": key, "mpid": this.mpid});
   }
+
+  /// Returns all GDPR Consent states with their purpose.
+  Future<Map> getGDPRConsentState() async {
+    String consentStateString =
+        await _channel.invokeMethod('getGDPRConsentState', {"mpid": this.mpid});
+    Map consentStateMap = jsonDecode(consentStateString);
+    return consentStateMap;
+  }
+
+  /// Set a GDPR consent object to a purpose on the user
+  ///
+  /// Sets a GDPR consent object [consent] to a purpose [purpose] on the user
+  void addGDPRConsentState(GDPRConsent consent, String purpose) async {
+    return await _channel.invokeMethod('addGDPRConsentState', {
+      'consented': consent.consented,
+      'document': consent.document,
+      'timestamp': consent.timestamp,
+      'location': consent.location,
+      'hardwareId': consent.hardwareId,
+      'purpose': purpose,
+      'mpid': this.mpid,
+    });
+  }
+
+  /// Remove a GDPR consent object from a purpose on the user
+  ///
+  /// Remove a GDPR consent object from a purpose [purpose] on the user
+  void removeGDPRConsentState(String purpose) async {
+    return await _channel.invokeMethod('removeGDPRConsentState', {
+      'purpose': purpose,
+      'mpid': this.mpid,
+    });
+  }
+
+  /// Returns the CCPA Consent state.
+  Future<Map> getCCPAConsentState() async {
```

Comment:
similar to above comment but instead of a map it should be just CCPAConsent since only 1 can exist 
```suggestion
  Future<CCPAConsent> getCCPAConsentState() async {
```